packages/cli/src/lib/bundler/scalprumConfig.ts

```diff
@@ -0,0 +1,160 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DynamicRemotePlugin } from '@openshift/dynamic-plugin-sdk-webpack';
+import ESLintPlugin from 'eslint-webpack-plugin';
+import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';
+import webpack, { ProvidePlugin } from 'webpack';
+
+import { resolve as resolvePath } from 'path';
+
+import { BundlingPaths } from './paths';
+import { transforms } from './transforms';
+import { DynamicPluginOptions } from './types';
+
+/**
+ * TODO: Create a config API to configure and optimize dependency sharing
+ * https://medium.com/@marvusm.mmi/webpack-module-federation-think-twice-before-sharing-a-dependency-18b3b0e352cb
+ */
+export const sharedModules = {
+  /**
+   * Mandatory singleton packages for sharing
+   */
+  react: {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  'react-dom': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  'react-router-dom': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  '@backstage/version-bridge': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  '@backstage/core-plugin-api': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  '@scalprum/react-core': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  '@openshift/dynamic-plugin-sdk': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  /**
+   * The following two packages are required to be shared as singletons to enable UI theming
+   */
+  '@material-ui/core/styles': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  '@material-ui/styles': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+};
+
+export async function createScalprumConfig(
+  paths: BundlingPaths,
+  options: DynamicPluginOptions,
+): Promise<webpack.Configuration> {
+  const { checksEnabled, isDev } = options;
+
+  const { plugins, loaders } = transforms({
+    ...options,
+    isDev: !!options.isDev,
+  });
+
+  if (checksEnabled) {
+    plugins.push(
+      new ForkTsCheckerWebpackPlugin({
+        typescript: { configFile: paths.targetTsConfig, memoryLimit: 4096 },
+      }),
+      new ESLintPlugin({
+        context: paths.targetPath,
+        files: ['**/*.(ts|tsx|mts|cts|js|jsx|mjs|cjs)'],
+      }),
+    );
+  }
+
+  // TODO(blam): process is no longer auto polyfilled by webpack in v5.
+  // we use the provide plugin to provide this polyfill, but lets look
+  // to remove this eventually!
+  plugins.push(
+    new ProvidePlugin({
+      process: 'process/browser',
+      Buffer: ['buffer', 'Buffer'],
+    }),
+  );
+
+  plugins.push(
+    new webpack.EnvironmentPlugin({
+      APP_CONFIG: options.frontendAppConfigs,
+    }),
+  );
+
+  const dynamicPluginPlugin = new DynamicRemotePlugin({
```

Comment:
This is mostly just a `ModuleFederationPlugin` right? Would it be possible to inline the relevant logic instead to keep this minimal and more direct towards the specific pieces that we're looking to implement here?

Reply:
It does a few more things. It generates the manifest, does validation of the inputs and adds a JSONp wrapper if a registration method of a plugins is set to be JSONp.